lib/auth/join_azure.go

```diff
@@ -372,6 +372,7 @@ func (a *Server) RegisterUsingAzureMethod(
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	joinRequest = req.RegisterUsingTokenRequest
```

Comment:
So before we were always calling `handleJoinFailure` with `nil`?

Also if I understand correctly, we don't log any information server-side if the challenge failes?

Reply:
Should this line be before the challenge block so we can log the details of the failed challenges?

Reply:
Unfortunately previous design has shot us in the foot here - we don't know the RegisterUsingTokenRequest before we've issued the challenge and received the response from the client. I have no clue why the client doesn't send this until we've issued the challenge /shrug

Reply:
😭 This makes troubleshooting failed joins way harder